embedded-hal-bus/Cargo.toml

```diff
@@ -26,7 +36,7 @@ embedded-hal = { version = "1.0.0", path = "../embedded-hal" }
 embedded-hal-async = { version = "1.0.0", path = "../embedded-hal-async", optional = true }
 critical-section = { version = "1.0" }
 defmt-03 = { package = "defmt", version = "0.3", optional = true }
-portable-atomic = {version = "1", default-features = false}
+portable-atomic = {version = "1.3", default-features = false, features = ["require-cas"]}
```

Comment:
doesn't this need `optional = true`? otherwise thumbv6 build will fail due to `require-cas` being enabled if the user doesn't enable any of the features for `portable-atomic`

Reply:
You are correct.
I've now made portable-atomic optional and done the cleanup required to handle building without it enabled.

I've tested the following on a thumbv6 target:
- feature disabled - build success, AtomicCell, {spi,i2c}::AtomicDevice not present in docs
- feature enabled, no polyfill set - compile error
- feature enabled, portable-atomic/unsafe-assume-single-core - build success, AtomicCell, {spi,i2c}::AtomicDevice present in docs